lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/TaskManager.cs

```diff
@@ -200,13 +200,13 @@ internal void Reset()
 
         /// <summary>
         /// This method is called when receiving ICompletedTask event during task running or system shutting down.
-        /// Removes the task from running tasks
+        /// Removes the task from running tasks if it was running
         /// Changes the task state from RunningTask to CompletedTask
         /// </summary>
         /// <param name="completedTask"></param>
         internal void RecordCompletedTask(ICompletedTask completedTask)
         {
-            RemoveRunningTask(completedTask.Id);
+            _runningTasks.Remove(completedTask.Id);
```

Comment:
@afchung , this is needed because there is a function below called CloseAllRunningTasks() which will send Close message to all the tasks and also removes taks from _runningTasks below. If @jwang98052  directly call RemoveRunningTask function she will get the exception..I am wondering if there is a cleaner way: Perhaps, do not remove from _runningTasks queue in CloseAllRunningTasks() and remove only once we receive failure or completed task messages. Problem  I faced while reviewing earlier also was that there were so many cases which makes me confused during reviewing. But I also do not have a cleaner solution :) .